test/expect-tests/fiber/fiber_tests.ml

```diff
@@ -847,6 +847,60 @@ let%expect_test "raise exception" =
        (fun () -> Pool.stop pool));
   [%expect {| Caught Exit |}]
 
+let%expect_test "double run a pool" =
+  (* Calling [Pool.run] on the same pool shouldn't be allowed *)
+  (Scheduler.run
+  @@
+  let pool = Pool.create () in
+  Fiber.fork_and_join_unit (fun () -> Pool.run pool) (fun () -> Pool.run pool));
+  [%expect.unreachable]
+  [@@expect.uncaught_exn {| (Test_scheduler.Never) |}]
+
+let%expect_test "start -> stop -> start a pool" =
+  (* We shouldn't be able to call [Pool.run] after stopping it *)
+  (Scheduler.run
+  @@
+  let pool = Pool.create () in
+  let* () =
+    Fiber.fork_and_join_unit
+      (fun () -> Pool.run pool)
+      (fun () -> Pool.stop pool)
+  in
+  Pool.run pool);
+  [%expect.unreachable]
+  [@@expect.uncaught_exn {| (Test_scheduler.Never) |}]
+
+let%expect_test "stop a pool and then run it" =
+  (* Weird deadlock when we call [stop] after [run] *)
+  (Scheduler.run
+  @@
+  let pool = Pool.create () in
+  let* () = Pool.stop pool in
+  let* () = Pool.run pool in
+  Pool.run pool);
```

Comment:
I don't understand: you again have a double `run` here, not a `stop` after `run` as the comment says.

Reply:
Ah yup, the 2nd run isn't necessary. The bug just needs stop and then run. Fixed.